tests/integration_tests/model_tests.py

```diff
@@ -57,30 +57,36 @@ def test_database_schema_presto(self):
         sqlalchemy_uri = "presto://presto.airbnb.io:8080/hive/default"
         model = Database(database_name="test_database", sqlalchemy_uri=sqlalchemy_uri)
 
-        db = make_url(model.get_sqla_engine().url).database
-        self.assertEqual("hive/default", db)
+        with model.get_sqla_engine_with_context() as engine:
+            db = make_url(engine.url).database
+            self.assertEqual("hive/default", db)
 
-        db = make_url(model.get_sqla_engine(schema="core_db").url).database
-        self.assertEqual("hive/core_db", db)
+        with model.get_sqla_engine_with_context(schema="core_db") as engine:
+            db = make_url(engine.url).database
+            self.assertEqual("hive/core_db", db)
 
         sqlalchemy_uri = "presto://presto.airbnb.io:8080/hive"
         model = Database(database_name="test_database", sqlalchemy_uri=sqlalchemy_uri)
 
-        db = make_url(model.get_sqla_engine().url).database
-        self.assertEqual("hive", db)
+        with model.get_sqla_engine_with_context() as engine:
+            db = make_url(engine.url).database
+            self.assertEqual("hive", db)
 
-        db = make_url(model.get_sqla_engine(schema="core_db").url).database
-        self.assertEqual("hive/core_db", db)
+        with model.get_sqla_engine_with_context(schema="core_db") as engine:
+            db = make_url(engine.url).database
+            self.assertEqual("hive/core_db", db)
 
     def test_database_schema_postgres(self):
         sqlalchemy_uri = "postgresql+psycopg2://postgres.airbnb.io:5439/prod"
         model = Database(database_name="test_database", sqlalchemy_uri=sqlalchemy_uri)
 
-        db = make_url(model.get_sqla_engine().url).database
-        self.assertEqual("prod", db)
+        with model.get_sqla_engine_with_context() as engine:
+            db = make_url(engine.url).database
```

Comment:
I see we have this `make_url_safe` here: `superset/databases/utils.py`, can we make use of that or we must go with this one?

Reply:
`make_url_safe` prevents password from showing up in the logs when `make_url` fails for some reason. It should always be used in production, but in the tests it's fine to use `make_url` instead.

Reply:
Good to know 👍 Thanks!